Jenkinsfile

```diff
@@ -34,34 +36,35 @@ def build(Build build) {
         build.builder.buildSteps.each {
           it()
         }
-        build.builder.postSteps.success.each {
-          it()
-        }
-        gitNotify ("New CI: " + build.name, "Finish", 'SUCCESS')
-      } catch(Exception e) {
         if (currentBuild.currentResult == 'SUCCESS') {
-            gitNotify ("New CI: " + build.name, "FAILURE", 'FAILURE')
-            print "Error: " + e
-            currentBuild.result = 'FAILURE'
-        }
-        else if(currentBuild.currentResult == 'UNSTABLE') {
-          build.builder.postSteps.unstable.each {
+          build.builder.postSteps.success.each {
             it()
           }
-        }
-        else if(currentBuild.currentResult == 'FAILURE') {
-          build.builder.postSteps.failure.each {
+        } else if(currentBuild.currentResult == 'UNSTABLE') {
+          build.builder.postSteps.unstable.each {
             it()
           }
         }
-        else if(currentBuild.currentResult == 'ABORTED') {
-          build.builder.postSteps.aborted.each {
-            it()
-          }
+      } catch(FlowInterruptedException e) {
+        print "Looks like we ABORTED"
+        currentBuild.result = 'ABORTED'
+        build.builder.postSteps.aborted.each {
+          it()
+        }
+      } catch(Exception e) {
+        print "Error was detected: " + e
+        currentBuild.result = 'FAILURE'
+        build.builder.postSteps.failure.each {
+          it()
         }
       }
       // ALWAYS
       finally {
+        if (currentBuild.currentResult == 'SUCCESS')
```

Comment:
What about reporting other statuses? What I was suggesting is to just execute:
`gitNotify ('JenkinsCI', build.name, currentBuild.currentResult)`

Reply:
The github status is different from Jenkins status

```
      Jenkins  : Github
               : "PENDING"
     "SUCCESS" : "SUCCESS",	
     "UNSTABLE":  	
     "ABORTED" :  
     "FAILURE" : "FAILURE",
               : "ERROR"
```
gitNotify (Name , Jenkins status or massage , Github Status )